mgmt_upgrade_test.py

```diff
@@ -0,0 +1,199 @@
+import logging
+from time import sleep
+
+from sdcm.tester import ClusterTester
+from sdcm.mgmt import get_scylla_manager_tool, TaskStatus, update_config_file
+from mgmt_cli_test import BackupFunctionsMixIn
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+class ManagerUpgradeTest(BackupFunctionsMixIn, ClusterTester):
+    DESTINATION = '/tmp/backup'
+    CLUSTER_NAME = "cluster_under_test"
+
+    def create_simple_table(self, table_name, keyspace_name="ks1"):
+        with self.cql_connection_patient(self.db_cluster.nodes[0]) as session:
+            session.execute(f"""CREATE KEYSPACE IF NOT EXISTS {keyspace_name}
+                WITH replication = {{'class': 'SimpleStrategy', 'replication_factor': '3'}};""")
+            session.execute(f"""CREATE table IF NOT EXISTS {keyspace_name}.{table_name} (
+                                key varchar PRIMARY KEY,
+                                c1 text,
+                                c2 text);""")
+
+    def write_multiple_rows(self, table_name, key_range, keyspace_name="ks1"):
+        with self.cql_connection_patient(self.db_cluster.nodes[0]) as session:
+            for num in range(*key_range):
+                session.execute(f"insert into {keyspace_name}.{table_name} (key, c1, c2) "
+                                f"VALUES ('k_{num}', 'v_{num}', 'v_{num}');")
+
+    def _create_and_add_cluster(self):
+        manager_node = self.monitors.nodes[0]
+        manager_tool = get_scylla_manager_tool(manager_node=manager_node)
+        self.update_all_agent_config_files()
+        current_manager_version = manager_tool.version
+        mgr_cluster = manager_tool.add_cluster(name=ManagerUpgradeTest.CLUSTER_NAME, db_cluster=self.db_cluster,
+                                               auth_token=self.monitors.mgmt_auth_token)
+        return mgr_cluster, current_manager_version
+
+    def test_upgrade(self):  # pylint: disable=too-many-locals,too-many-statements
+        target_upgrade_version = self.params.get('new_scylla_mgmt_repo')
+        manager_node = self.monitors.nodes[0]
+        manager_tool = get_scylla_manager_tool(manager_node=manager_node)
+        manager_tool.add_cluster(name="cluster_under_test", db_cluster=self.db_cluster,
+                                 auth_token=self.monitors.mgmt_auth_token)
+        current_manager_version = manager_tool.version
+
+        LOGGER.debug("Generating load")
+        self.generate_load_and_wait_for_results()
+
+        mgr_cluster = manager_tool.get_cluster(cluster_name="cluster_under_test")
+
+        with self.subTest("Creating reoccurring backup and repair tasks"):
+
+            repair_task = mgr_cluster.create_repair_task(interval="1d")
+            repair_task_current_details = wait_until_task_finishes_return_details(repair_task)
+
+            self.update_all_agent_config_files()
+            bucket_name = self.params.get('backup_bucket_location').split()[0]
```

Comment:
why do you split it?

Reply:
Completely honestly, because Fabio created the param and this is how he used it. I think the plan is that at some point there will be multiple buckets, but I'm not sure. @fgelcer ?

Reply:
because if we run with a multi DC setup, it will be a list of `backup_bucket_location`, one for each DC (they are in a kind of yaml format, like `bucket1 bucket2 bucket3`...

also, i think the manager API allow to send a list of locations, exactly for multi DC